fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/EquationRewriter.java

```diff
@@ -169,9 +189,167 @@ public ScalarOperator visitCall(CallOperator call, Void context) {
+                        return null;
+                    }
+                    return new CallOperator(FunctionSet.IF, newFn.getReturnType(), args, newFn);
+                } else if (arg0Call instanceof CaseWhenOperator) {
+                    CaseWhenOperator caseClause = (CaseWhenOperator) arg0Call;
+                    // if case condition contains any agg column ref, return false
+                    ScalarOperator caseExpr = caseClause.hasCase() ? caseClause.getCaseClause() : null;
+                    if (caseExpr != null && isContainAggregateColumn(caseExpr, operatorMap)) {
+                        return null;
+                    }
+                    List<ScalarOperator> newCaseWhens = Lists.newArrayList();
+                    for (int i = 0; i < caseClause.getWhenClauseSize(); i++) {
+                        ScalarOperator when = caseClause.getWhenClause(i);
+                        if (isContainAggregateColumn(when, operatorMap)) {
+                            return null;
+                        }
+                        newCaseWhens.add(when);
+
+                        // when clause or else clause can only contain aggregate column ref
+                        ScalarOperator then = caseClause.getThenClause(i);
+                        ScalarOperator newThen = rewriteIfOnlyContianAggregateColumn(then, operatorMap);
+                        if (newThen == null) {
+                            return null;
+                        }
+                        newCaseWhens.add(newThen);
+                    }
+                    ScalarOperator elseClause = caseClause.hasElse() ? caseClause.getElseClause() : null;
+                    ScalarOperator newElseClause = elseClause;
+                    if (elseClause != null) {
+                        newElseClause = rewriteIfOnlyContianAggregateColumn(elseClause, operatorMap);
+                        if (newElseClause == null) {
+                            return null;
+                        }
+                    }
+                    // NOTE: use call's result type as its input.
+                    return new CaseWhenOperator(call.getType(), caseExpr, newElseClause, newCaseWhens);
+                }
+            }
+            return null;
+        }
+
+        private boolean isContainAggregateColumn(ScalarOperator child,
+                                                 Map<ColumnRefOperator, CallOperator> operatorMap) {
+            return child.getColumnRefs().stream().anyMatch(x -> operatorMap.containsKey(x));
+        }
+
+        private ScalarOperator rewriteIfOnlyContianAggregateColumn(ScalarOperator child,
+                                                                   Map<ColumnRefOperator, CallOperator> operatorMap) {
+            List<ColumnRefOperator> colRefs = child.getColumnRefs();
+            if (colRefs.size() > 1) {
+                return null;
+            }
+            // constant operator
+            if (colRefs.size() == 0) {
```

Comment:
how about const null operator?
sum(null) seems should be rewritten into 0;
max(null) seems should be rewritten into null;